Objects/floatobject.c

```diff
@@ -2048,14 +2048,13 @@ PyFloat_Pack2(double x, char *data, int le)
         bits = 0;
     }
     else if (isnan(x)) {
-        /* There are 2046 distinct half-precision NaNs (1022 signaling and
-           1024 quiet), but there are only two quiet NaNs that don't arise by
-           quieting a signaling NaN; we get those by setting the topmost bit
-           of the fraction field and clearing all other fraction bits. We
-           choose the one with the appropriate sign. */
         sign = (copysign(1.0, x) == -1.0);
         e = 0x1f;
-        bits = 512;
+
+        uint64_t v;
+
+        memcpy(&v, &x, sizeof(v));
+        bits = (v & 0xffc0000000000ULL) >> 42; /* NaN's payload */
```

Comment:
There are warnings:

'=': conversion from 'uint64_t' to 'unsigned short', possible loss of data [D:\a\cpython\cpython\PCbuild\_freeze_module.vcxproj]